fraqbot/Local/randomemoji.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+import random
+import sys
+
+from Legobot.Lego import Lego
+
+
+logger = logging.getLogger(__name__)
+LOCAL_DIR = os.path.join(os.path.abspath(os.path.dirname(__file__)))
+
+if LOCAL_DIR not in sys.path:
+    sys.path.append(LOCAL_DIR)
+
+
+from helpers import utils  # noqa: E402
+
+
+class RandomEmoji(Lego):
+    def __init__(self, baseplate, lock, *args, **kwargs):
+        super().__init__(baseplate, lock, acl=kwargs.get('acl'))
+
+    def listening_for(self, message):
+        text = message.get('text')
+
+        return isinstance(text, str) and text.startswith('!emoji')
+
+    def _get_emoji(self, how_many):
+        maximum_how_many = 20
+        minimum_how_many = 1
+        how_many_limited = max(minimum_how_many, min(maximum_how_many, how_many))
+        
+        emoji_response = utils.call_slack_api(
+                self.slack_client,
+                'emoji.list',
+                True,
+                'emoji',
+                total_limit=10000,
+                limit=1000,
+            )
+        
+        final_emoji_text = ""
+
+        x = range(int(how_many_limited))
+        for n in x:
+            final_emoji_text += ' :' + random.choice(list(emoji_response.items()))[0] + ':'
```

Comment:
A few things here, both for learning and for changes:

- `how_man_limited` should already be an int, so we don't need to cast it again
- we don't need to assign the range iterator to a variable, we can just do `for n in range...`
- ultimately we don't need the iterator anyway, as I believe you can just use random.choices: https://docs.python.org/3/library/random.html#random.choices
- `list(emoji_response.items()))[0]` is pretty inefficient. Let's improve on it.
  - First, since you're not using the values only the keys you can use `.keys()` instead of `.items()` and then you don't have to select the first item in the resulting tuple
  - Second, running `list( .items())` over and over is slow and resource consuming. Why not assign it to a list first and reuse it? This will be especially helpful for using `random.choices` instead of `random.choice`